tests/integration/src/test/scala/mesosphere/marathon/integration/ExternalVolumesIntegrationTest.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.marathon
+package integration
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.integration.setup._
+import play.api.libs.json.Json
+
+class ExternalVolumesIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
```

Comment:
This is the piece that was missing in #7024 – an integration test that verifies we can actually create an app with an external volume that serializes options into its name.

Reply:
Sigh. We shouldn't an integration test for a validation change.

Reply:
We have a unit test for AppsResource and PodsResource which passes through the full validation chain. I think that would be lighter and sufficient

Reply:
Can we please instead just make this an AppsResourceTest ?

Reply:
I'll ignore the _sigh_ assuming you didn't mean this in any negative way :)

I changed this to an AppsResourceTest – but only because I had the integration test in between to verify this actually works. I want to be very clear here: we pushed out a release not realizing we had another layer of validation in here. We had limited test coverage with unit tests that seemed fine, and had no end-to-end test whatsoever.

So, sure, a unit test is sufficient to prevent a regression. Still an end-to-end test is required to actually test the feature :)